File: prid1920-g13/Models/Post.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid_1819_g13.Models
{
    
    public class Post
    {
        [Key]
        public int Id { get; set; }
        public string Title { get; set; }
        [Required(ErrorMessage = "Required")]
        public string Body { get; set; }
        [Required(ErrorMessage = "Required")]
        public DateTime Timestamp { get; set; } = DateTime.Now;
        public int AuthorId { get; set; }
        public int? ParentId { get; set; }
        public int? AcceptedPostId { get; set; }
        public virtual  User User { get; set; }
        public virtual  Post ParentPost { get; set; }
        public virtual Post AcceptedPost { get; set; }
        public virtual IList<Vote> Votes { get; set; } = new List<Vote>();
        public virtual IList<Post> Reponses { get; set; } = new List<Post>();
        public virtual List<PostTag> PostTags { get; set; } = new List<PostTag>();
        public virtual IList<Comment> Comments { get; set; } = new List<Comment>();
        [NotMapped]
        public int Score 
        {
            get => Votes.Sum(v => v.UpDown);
        }
        [NotMapped]
        public IEnumerable<Tag> Tags
        {
            get =>PostTags.Select(f => f.Tag);
        }
        [NotMapped]
        public int MaxScore 
        {
            get => Reponses.Count() > 0 ? Reponses.Max(p => p.Score) > this.Score ?  Reponses.Max(p => p.Score) : this.Score : 0 ;
        }

    }
}

File: prid1920-g13/ClientApp/src/app/models/Post.ts
import { User } from "./User";
import {UserService} from '../services/user.service';
import { Tag } from "./Tag";
import { Vote } from "./Vote";
export class Post {
    public id: number;
    public title: string;
    public body: string;
    public timestamp: Date;
    public reponses: Post[];
    public tags: Tag[];
    public user: User;
    public score: number;
    public comments: any;
    public acceptedPostId: number;
    public parentId: number;
    public votes: Vote[];
    public maxScore: number;
    public numUp: number;
    public numDown: number;
    constructor(data: any) {
        if (data) {
            this.id = data.id,
            this.title = data.title,
            this.body = data.body,
            this.timestamp = data.timestamp,
            this.reponses = data.reponses,
            this.tags = data.tags,
            this.user = data.user,
            this.score = data.score,
            this.comments = data.comments,
            this.acceptedPostId = data.acceptedRepId
            this.parentId = data.parentId,
            this.votes = data.votes,
            this.maxScore = data.maxScore,
            this.numUp = data.numUp,
            this.numDown = data.numDown
        }
    }
}